src/Stratis.Bitcoin.Features.Api/MvcBuilderExtensions.cs

```diff
@@ -13,13 +15,22 @@ public static class MvcBuilderExtensions
         /// </summary>
         /// <param name="builder">The builder</param>
         /// <param name="features">The selected features to include the assemblies of.</param>
+        /// <param name="serviceCollection">The full node services.</param>
         /// <returns>The Mvc builder</returns>
-        public static IMvcBuilder AddControllers(this IMvcBuilder builder, IEnumerable<IFullNodeFeature> features)
+        public static IMvcBuilder AddControllers(this IMvcBuilder builder, IEnumerable<IFullNodeFeature> features, IServiceCollection serviceCollection)
         {
-            // The required assemblies including those of the selected features.
+            // Detect explicitly registered transient controllers.
+            // This supports the scenario where a controller is registered by a different feature than the one it belongs to.
+            List<Type> explicitControllers = serviceCollection
+                .Where(s => s.Lifetime == ServiceLifetime.Transient)
```

Comment:
Do we need to care about the lifetime here?

Reply:
Controllers should only ever be registered as transient. This would help discourage anything else?